src/Utilities/QuaternionHelpers.hpp

```diff
@@ -0,0 +1,33 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines helper functions for converting between `std::array` and boost
+/// quaternions.
+
+#pragma once
+
+#include <boost/math/quaternion.hpp>
+
+template <typename T>
+std::array<T, 4> boostquat_to_array(const boost::math::quaternion<T>& input) {
+  return std::array<T, 4>{input.R_component_1(), input.R_component_2(),
+                          input.R_component_3(), input.R_component_4()};
+}
+
+template <typename T>
+boost::math::quaternion<T> array_to_boostquat(const std::array<T, 4>& input) {
+  return boost::math::quaternion<T>(input[0], input[1], input[2], input[3]);
+}
+
+template <typename T>
+boost::math::quaternion<T> threearray_to_boostquat(
+    const std::array<T, 3>& input) {
+  return boost::math::quaternion<T>(static_cast<T>(0), input[0], input[1],
+                                    input[2]);
+}
+
+template <typename T>
+void make_unit_quaternion(boost::math::quaternion<T>& input) {
```

Comment:
We don't generally allow mutable references as function arguments.  Either take the argument as `const gsl::not_null<boost::math::quaternion<T>*>` (from `Utilities/Gsl.hpp`) or return a new quaternion instead of modifying the argument.